File: week4/java/week4_9.java
import java.util.Scanner;

public class week4_9 {
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();

        int cur = 1;
        int answer = 0;
        while(cur<n){
            int sum = cur;
            String curStr = String.valueOf(cur);

            for(int i=0; i<curStr.length(); i++){
                sum += curStr.charAt(i)-'0';
            }

            if(sum == n){
                answer = cur;
                break;
            }

            cur++;
        }

        System.out.println(answer);
    }
}

File: week4/c++/week4_9.cpp
#include <iostream>
#include <string>

using namespace std;

int main(void)
{
    int n;
    cin >> n;

    int answer = 0;
    for (int i = 1; i <= n; i++)
    {

        int sum = 0;
        string str = to_string(i);
        for (int k = 0; k < str.length(); k++)
        {
            sum += str[k] - 48;
        }
        sum += i;

        if (sum == n)
        {
            answer = i;
            break;
        }
    }

    cout << answer << endl;

    return 0;
}